src/Compilers/CSharp/Test/Symbol/Symbols/GenericConstraintTests.cs

```diff
@@ -6862,5 +6862,190 @@ static void Main(string[] args)
+        public void Bug41779_Class()
+        {
+            var source =
+@"using System;
+class C
+{
+    String GetService(){ return String.Empty; }
+}
+
+static class Program
+{
+    static T GetService<T>(this C obj) => default;
+    
+    static void M(C provider)
+    {
+        provider.GetService<>();
+        provider.GetService<>().ToString();
+        provider.GetService<>();
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (13,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(13, 9),
+                // (14,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>().ToString();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(14, 9),
+                // (15,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(15, 9)
+            );
+        }
+
+        [Fact]
+        [WorkItem(41779, "https://github.com/dotnet/roslyn/issues/41779")]
+        public void Bug41779_ClassAbstract()
+        {
+            var source =
+@"using System;
+abstract class C
+{
+    String GetService(){ return String.Empty; }
+}
+
+static class Program
+{
+    static T GetService<T>(this C obj) => default;
+    
+    static void M(C provider)
+    {
+        provider.GetService<>();
+        provider.GetService<>().ToString();
+        provider.GetService<>();
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (13,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(13, 9),
+                // (14,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>().ToString();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(14, 9),
+                // (15,9): error CS0305: Using the generic method group 'GetService' requires 1 type arguments
+                //         provider.GetService<>();
+                Diagnostic(ErrorCode.ERR_BadArity, "provider.GetService<>").WithArguments("GetService", "method group", "1").WithLocation(15, 9)
+            );
+        }
```

Comment:
> } [](start = 7, length = 2)

I recommend splitting test scenarios in three category:
1) binding to an instance method
2) binding to an extension method
3) binding to a static method

Test all of them separately for generic and non-generic cases.

Reply:
If a combination of both an instance and an extension method is needed in order to observe the problem, then we need to keep tests like that. However, I would like us to get a better understanding what kind of interaction is going on that causes us to fail in a scenario like that. The error messages indicate that we bound to the extension method. Why would presence of some other method cause us to fail to report an error.

---
In reply to: [442351010](https://github.com/dotnet/roslyn/pull/45033#discussion_r442351010) [](ancestors = 442351010)